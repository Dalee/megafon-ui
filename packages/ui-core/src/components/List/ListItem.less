@import '~styles/base.less';

.mfui-beta-list-item {
    margin: 12px 0 12px 40px;
}
