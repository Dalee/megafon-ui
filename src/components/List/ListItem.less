@import '~styles/base.less';

.list-item {
    margin: 12px 0px 12px 40px;
}
