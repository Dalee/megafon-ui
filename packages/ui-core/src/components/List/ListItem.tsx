import * as React from 'react';
import * as PropTypes from 'prop-types';
import './ListItem.less';
import cnCreate from 'utils/cn';

interface IListItemProps {
    /** Custom classname */
    className?: string;
    /** Left margin */
    disableLeftMargin?: boolean;
    /** Children */
    children?: React.ReactNode;
}

const cn = cnCreate('mfui-list-item');
class ListItem extends React.Component<IListItemProps, {}> {
    static propTypes = {
        className: PropTypes.string,
        disableLeftMargin: PropTypes.bool,
        children: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.element),
            PropTypes.arrayOf(PropTypes.node),
            PropTypes.element,
            PropTypes.string,
            PropTypes.node,
        ]),
    };

    static defaultProps = {
        disableLeftMargin: false,
    };

    render() {
        const {className, disableLeftMargin, children } = this.props;

        return (
            <li className={cn('', {'disable-left-margin': disableLeftMargin}, className)}>
                {children}
            </li>
        );
    }
}

export default ListItem;
