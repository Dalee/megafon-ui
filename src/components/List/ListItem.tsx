import * as React from 'react';
import * as PropTypes from 'prop-types';
import './ListItem.less';
import { cnCreate } from '../../utils/cn';

interface Props {
    children?: JSX.Element[] | Element[] | JSX.Element | string | Element;
}

const cn = cnCreate('list-item');
class ListItem extends React.Component<Props, {}> {
    static propTypes = {
        children: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.element),
            PropTypes.arrayOf(PropTypes.node),
            PropTypes.element,
            PropTypes.string,
            PropTypes.node,
        ]),
    };

    render() {
        return (
            <li className={cn('')}>
                {this.props.children}
            </li>
        );
    }
}

export default ListItem;
