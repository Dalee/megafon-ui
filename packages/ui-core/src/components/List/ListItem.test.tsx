import * as React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';

describe('<ListItem />', () => {
    it('it renders ListItem', () => {
        const wrapper = shallow(<ListItem>item</ListItem>);
        expect(wrapper).toMatchSnapshot();
    });

    it('it render with disable margin left', () => {
        const wrapper = shallow(<ListItem disableLeftMargin>item</ListItem>);
        expect(wrapper).toMatchSnapshot();
    });
});
