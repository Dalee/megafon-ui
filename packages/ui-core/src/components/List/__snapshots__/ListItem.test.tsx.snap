// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> it renders ListItem 1`] = `
<li
  className="mfui-beta-list-item"
>
  item
</li>
`;
